tls/s2n_handshake_io.c

```diff
@@ -668,20 +672,20 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
         return S2N_SUCCESS;
     }
 
-    S2N_ERROR_IF(conn->handshake.handshake_type & HELLO_RETRY_REQUEST, S2N_ERR_INVALID_HELLO_RETRY);
+    GUARD_AS_POSIX(s2n_handshake_type_reset(conn));
```

Comment:
How are these equivalent?

Reply:
They're not, they just happen to be in the same location.

I was thinking that we can't do the HELLO_RETRY_REQUEST check anymore, because if we're in this method then we're using TLS1.2 and the HELLO_RETRY_REQUEST flag doesn't really exist. If the 0x10 flag is set, that could either mean HELLO_RETRY_REQUEST was illegally set or TLS12_PERFECT_FORWARD_SECRECY was illegally set.

I guess I could add a sanity check that handshake_type==0 to detect any unexpected changes to the handshake_type before this method, but we didn't do that before and the s2n_handshake_type_reset call resets the handshake_type to zero anyway.